app/controllers/actions/AuthAction.scala

```diff
@@ -60,3 +70,12 @@ trait AuthAction
     extends ActionBuilder[AuthenticatedRequest, AnyContent] with ActionFunction[Request, AuthenticatedRequest]
 
 case class NoExternalId() extends NoActiveSession("No externalId was found")
+@ProvidedBy(classOf[EoriWhitelistProvider])
+class EoriWhitelist(values: Seq[String]) {
+  def allow(user: SignedInUser): Boolean = values.isEmpty || values.contains(user.eori)
+}
+
+class EoriWhitelistProvider @Inject()(configuration: Configuration) extends Provider[EoriWhitelist] {
```

Comment:
Wouldn't we normally inject our AppConfig class (rather than Configuration) and leave that class to lazy load the config values (i.e. eori list)?

Reply:
Approach with injecting `Configuration` give us possibility to split huge `AppConfig` to few different for specific Configuration. So for whitelisting we will have one small Provider

Reply:
Moreover what you can see here, we don't need anything from `AppConfig` in `AuthAction`, just information about eori, so instead of injecting huge class we can inject this and have access to just one configuration

Reply:
Personally, I don't see `AppConfig` being a problem, it's nice being able to look in once place for all configuration

Reply:
`AppConfig` becomes [God Object](https://en.wikipedia.org/wiki/God_object) for system configuration. Every time you have use it it ends with mock or stub that configure half of system. Or become mocking `AppConfig` itself and to be honest i don't know, which is worse.

Generally if i would like to use `AppConfig` from level of provider there would be only to access configuration. Injecting object to access other object is quite [serious code smell (section digging into collabolators)](http://misko.hevery.com/code-reviewers-guide/)   On the other hand i can put new keys into `AppConfig` but this would break [SRP of `AppCofnig`](https://en.wikipedia.org/wiki/Single_responsibility_principle) and now we have multiple purpose why we have to change `AppConfig`. Using `AppConfig` would also violate [Demeter Law](https://en.wikipedia.org/wiki/Law_of_Demeter) expose internal structure and clutter this class into multiple place in code.

Here i don't have to use `ServicesConfig` and i don't see pros for using here `AppConfig` - it would just put more responsibilities on single object, which is now quite big and hard to maintain now, ultimately make it harder to test.

Reply:
My proposal is:

```
class EoriWhitelist @Inject()(configuration: Configuration) {
  lazy val values = configuration.get[Seq[String]]("whitelist.eori")
  def contains(user: SignedInUser): Boolean = values.isEmpty || values.contains(user.eori)
}
```